auth-server/src/services/auth.ts

```diff
@@ -0,0 +1,126 @@
+import * as jwt from 'jsonwebtoken'
+import { NotFoundError } from 'objection'
+import { DataError } from 'objection-db-errors'
+import { uuid } from 'uuidv4'
+
+import User from '../model/User'
+import RefreshToken from '../model/RefreshToken'
+
+export default class AuthService {
+	constructor(){}
+
+	public async Login(username, password): Promise<any> {
+		const user = await User
+			.query()
+			.where('username', username)
+			.first()
+      
+		if (!user) { 
+			throw new NotFoundError('User not found')
+		}
+
+		const correctPassword = await user.verifyPassword(password)
+		if (!correctPassword) {
+			throw new DataError('Incorrect password')
+		}
+
+		return {
+			user: {
+				email: user.email,
+				username: user.username,
+				name: user.name
+			},
+			token: this.getSignedToken(user),
+			refreshToken: await this.getRefreshToken(user)
+		}
+	}
+
+	public async SignUp(email, password, username, name): Promise<any> {
+		const user = await User.query()
+			.allowInsert('[email, password, username, name]')
+			.insert({
+				email,
+				password,
+				username,
+				name
+			})
+		
+		return {
+			user: {
+				email: user.email,
+				username: user.username,
+				name: user.name
+			},
+			token: this.getSignedToken(user),
+			refreshToken: await this.getRefreshToken(user)
+		}
+	}
+
+	public async RefreshToken(token: string) {
+		const refreshToken = await RefreshToken
+			.query()
+			.where('token', token)
+			.first()
+		
+		if (!refreshToken) { 
+			throw new NotFoundError('Refresh token not found')
+		}
+
+		if (!refreshToken.valid) {
+			throw new Error('Refresh token not valid')
+		}
+
+		if (refreshToken.expires < Math.floor(Date.now() / 1000)) {
+			throw new Error('Refresh token expired')
+		}
+
+		const user = await User
+			.query()
+			.where('id', refreshToken.user_id)
+			.first()
+
+		return this.getSignedToken(user)
+	}
+
+	private getSignedToken({ id, username, email }) {
+		const allowedRoles = ['user']
+		let currentRole = 'user'
+	
+		// if our user is the proposal bot, give additional role.
+		if (id == process.env.BOT_PROPOSAL_USER_ID) { // eslint-disable-line
```

Comment:
Just curious, doesn't eslint like env variables?

Reply:
This is because of == environment variable might be string so == is a correct check here